ledger/participant-integration-api/src/main/scala/platform/store/backend/StorageBackend.scala

```diff
@@ -283,9 +287,13 @@ object DBLockStorageBackend {
 }
 
 object StorageBackend {
-  case class Params(ledgerEnd: Offset, eventSeqId: Long)
-
-  case class LedgerEnd(lastOffset: Option[Offset], lastEventSeqId: Option[Long])
+  case class LedgerEnd(lastOffset: Offset, lastEventSeqId: Long)
+  case class OptionalLedgerEnd(lastOffset: Option[Offset], lastEventSeqId: Option[Long])
+  case class IdentityParams(ledgerId: LedgerId, participantId: ParticipantId)
+  case class OptionalIdentityParams(
```

Comment:
> Simillarily here it seems not necessary since initialization always sets up both.

Same as above, it's not obvious that existing databases will always have both columns defined. We could make the participantId `NOT NULL` (same as the ledger id) and fix missing values in a migration to enforce this. This would clean up some code.

Reply:
Simillarily as above, the idea would be to hide the special treatments of one set and the other not in implementation, and not letting this leak out on the interface level. 
But no need in this case, I am ok with how it is right now